lib/core/versions/latest/BatchWriter.ts

```diff
@@ -77,4 +83,10 @@ export default class BatchWriter implements IBatchWriter {
     // Remove written operations from queue after batch writing has completed successfully.
     await this.operationQueue.dequeue(queuedOperations.length);
   }
+
+  private getNumberOfOpsToWrite (valueTimeLock: ValueTimeLockModel | undefined, normalizedFee: number): number {
+    const numberOfOpsAllowed = ValueTimeLockVerifier.calculateMaxNumberOfOpsAllowed(valueTimeLock, normalizedFee);
```

Comment:
Consider just let `calculateMaxNumberOfOpsAllowed` perform the `Math.min()` check.

Reply:
I want to keep the ValueTimeLockVerifier to only care about the lock related checks. The `maxOperationsPerBatch` is a protocol check which belongs outside of the verifier IMO. I am refactoring the code to make it more clearer.